ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/state/PendingBlocksManagerTest.java

```diff
@@ -259,4 +260,62 @@ public void shouldReturnLowestBlockByNumber() {
 
     assertThat(pendingBlocksManager.lowestAnnouncedBlock()).contains(parentBlock.getHeader());
   }
+
+  @Test
+  public void shouldReturnLowestAncestorPendingBlock() {
+    final BlockDataGenerator gen = new BlockDataGenerator();
+    final Block parentBlock = gen.block();
+
+    final Block childBlock = gen.nextBlock(parentBlock);
+    final Block childBlock2 = gen.nextBlock(childBlock);
+    final Block childBlock3 = gen.nextBlock(childBlock2);
```

Comment:
I'd be tempted to add a second fork in here so the test is actually ensuring it follows the specific fork. Currently it would pass just by returning the pending block with the lowest height.

Reply:
Done!